lib/SparkFun_AS7265x/SparkFun_AS7265X.cpp

```diff
@@ -0,0 +1,569 @@
+
+//Constructor
+AS7265X::AS7265X()
+{
+
+}
+
+//Initializes the sensor with basic settings
+//Returns false if sensor is not detected
+boolean AS7265X::begin(TwoWire &wirePort)
+{
+  _i2cPort = &wirePort;
+  _i2cPort->begin(); //This resets any setClock() the user may have done
+
+  if (isConnected() == false) return (false); //Check for sensor presence
+
+  //Check to see if both slaves are detected
+  uint8_t value = virtualReadRegister(AS7265X_DEV_SELECT_CONTROL);
+  if ( (value & 0b00110000) == 0) return (false); //Test if Slave1 and 2 are detected. If not, bail.
+
+  setBulbCurrent(AS7265X_LED_CURRENT_LIMIT_12_5MA, AS7265x_LED_WHITE);
+  setBulbCurrent(AS7265X_LED_CURRENT_LIMIT_12_5MA, AS7265x_LED_IR);
+  setBulbCurrent(AS7265X_LED_CURRENT_LIMIT_12_5MA, AS7265x_LED_UV);
+
+  disableBulb(AS7265x_LED_WHITE); //Turn off bulb to avoid heating sensor
+  disableBulb(AS7265x_LED_IR);
+  disableBulb(AS7265x_LED_UV);
+
+  setIndicatorCurrent(AS7265X_INDICATOR_CURRENT_LIMIT_8MA); //Set to 8mA (maximum)
+  enableIndicator();
+
+  setIntegrationCycles(49); //50 * 2.8ms = 140ms. 0 to 255 is valid.
+  //If you use Mode 2 or 3 (all the colors) then integration time is double. 140*2 = 280ms between readings.
+
+  setGain(AS7265X_GAIN_64X); //Set gain to 64x
+
+  setMeasurementMode(AS7265X_MEASUREMENT_MODE_6CHAN_ONE_SHOT); //One-shot reading of VBGYOR
+  
+  enableInterrupt();
+
+  return (true); //We're all setup!
+}
+
+uint8_t AS7265X::getDeviceType()
+{
+  return(virtualReadRegister(AS7265X_HW_VERSION_HIGH));
+}
+uint8_t AS7265X::getHardwareVersion()
+{
+  return(virtualReadRegister(AS7265X_HW_VERSION_LOW));
+}
+
+uint8_t AS7265X::getMajorFirmwareVersion()
+{
+  virtualWriteRegister(AS7265X_FW_VERSION_HIGH, 0x01); //Set to 0x01 for Major
+  virtualWriteRegister(AS7265X_FW_VERSION_LOW, 0x01); //Set to 0x01 for Major
+  
+  return(virtualReadRegister(AS7265X_FW_VERSION_LOW));
+}
+
+uint8_t AS7265X::getPatchFirmwareVersion()
+{
+  virtualWriteRegister(AS7265X_FW_VERSION_HIGH, 0x02); //Set to 0x02 for Patch
+  virtualWriteRegister(AS7265X_FW_VERSION_LOW, 0x02); //Set to 0x02 for Patch
+  
+  return(virtualReadRegister(AS7265X_FW_VERSION_LOW));
+}
+
+uint8_t AS7265X::getBuildFirmwareVersion()
+{
+  virtualWriteRegister(AS7265X_FW_VERSION_HIGH, 0x03); //Set to 0x03 for Build
+  virtualWriteRegister(AS7265X_FW_VERSION_LOW, 0x03); //Set to 0x03 for Build
+  
+  return(virtualReadRegister(AS7265X_FW_VERSION_LOW));
+}
+
+//Returns true if I2C device ack's
+boolean AS7265X::isConnected()
+{
+  _i2cPort->beginTransmission((uint8_t)AS7265X_ADDR);
+  if (_i2cPort->endTransmission() != 0)
+    return (false); //Sensor did not ACK
+  return (true);
+}
+
+//Tells IC to take all channel measurements and polls for data ready flag
+void AS7265X::takeMeasurements()
+{
+  setMeasurementMode(AS7265X_MEASUREMENT_MODE_6CHAN_ONE_SHOT); //Set mode to all 6-channels, one-shot
+
+  //Wait for data to be ready
+  while (dataAvailable() == false) delay(AS7265X_POLLING_DELAY);
```

Comment:
Here is the culprit.
In the plugin code you can check this in the `PLUGIN_TEN_PER_SECOND` or even `PLUGIN_FIFTY_PER_SECOND` and if it is done, read the values and reschedule `PLUGIN_READ` to return `success = true;` as you've new values.